ui/src/main/java/com/hyperwallet/android/ui/view/widget/DateUtil.java

```diff
@@ -0,0 +1,134 @@
+                DateFormat.getBestDateTimePattern(Locale.getDefault(), WIDGET_DATE_PATTERN), Locale.getDefault());
+    }
+
+    /**
+     * Convert Date from server format (yyyy-MM-dd) to DateWidget format build by BestDateTimePattern @see {@link
+     * DateFormat}  or (dd MMMM yyyy)
+     *
+     * @param serverDate Date from server
+     * @return String date in the format by BestDateTimePattern or dd MMMM yyyy otherwise
+     */
+    @NonNull
+    String convertDateFromServerToWidgetFormat(@Nullable final String serverDate) {
+        if (isServerDateNotValid(serverDate)) {
+            return "";
+        }
+        try {
+            final Calendar calendar = Calendar.getInstance();
+            calendar.setTime(mServerDateFormat.parse(serverDate));
+            return mWidgetDateFormat.format(calendar.getTime());
+        } catch (NumberFormatException | ParseException | IndexOutOfBoundsException e) {
+            return "";
+        }
+    }
+
+    /**
+     * Build Date in the server format (yyyy-MM-dd)
+     *
+     * @param year       the selected year
+     * @param month      the selected month (0-11 for compatibility with {@link Calendar#MONTH})
+     * @param dayOfMonth th selected day of the month (1-31, depending on month)
+     * @return String date in the format (yyyy-MM-dd)
+     */
+    @NonNull
+    String buildDateFromDateDialogToServerFormat(final int year, final int month, final int dayOfMonth) {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.set(year, month, dayOfMonth);
+        return mServerDateFormat.format(calendar.getTime());
+    }
+
+    /**
+     * Build Date in the DateWidget format build by BestDateTimePattern @see{@link DateFormat} or (dd MMMM yyyy)
+     *
+     * @param year       the selected year
+     * @param month      the selected month (0-11 for compatibility with {@link Calendar#MONTH})
+     * @param dayOfMonth th selected day of the month (1-31, depending on month)
+     * @return String date in the format @see{@link DateFormat} or (dd MMMM yyyy)
+     */
+    @NonNull
+    String buildDateFromDateDialogToWidgetFormat(final int year, final int month, final int dayOfMonth) {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.set(year, month, dayOfMonth);
+        return mWidgetDateFormat.format(calendar.getTime());
+    }
+
+    /**
+     * Convert Date from server format (yyyy-MM-dd) to Calendar
+     *
+     * @param serverDate Date from server
+     * @return Calendar with date from server
+     */
+    @NonNull
+    Calendar convertDateFromServerFormatToCalendar(@Nullable final String serverDate) {
+        final Calendar calendar = Calendar.getInstance();
+        if (isServerDateNotValid(serverDate)) {
+            return calendar;
+        }
+
+        try {
+            calendar.setTime(mServerDateFormat.parse(serverDate));
+            return calendar;
+        } catch (ParseException e) {
```

Comment:
if we don't care about the exception why not throw it back to the caller maybe it will make more sense to process that there? if not why not clean the code like so
```
    Calendar convertDateFromServerFormatToCalendar(@Nullable final String serverDate) {
        final Calendar calendar = Calendar.getInstance();
        if (isServerDateValid(serverDate)) {
            try {
                calendar.setTime(mServerDateFormat.parse(serverDate));
            } catch (ParseException e) { // will not process exception and defaults calendar to its own current instances' time and date }
        }
        return calendar;
    }

Reply:
Created DateParseException